src/client/src/rt-components/tear-off/ExternalWindow.tsx

```diff
@@ -12,51 +11,46 @@ const defaultConfig: WindowConfig = {
 }
 
 export interface ExternalWindowProps {
-  title: string
-  onBlock: () => void
+  title?: string
+  onBlock?: () => void
   onUnload: () => void
-  config: WindowConfig
+  config?: WindowConfig
 }
 
-const defaultWindowProps: ExternalWindowProps = {
-  title: '',
-  onBlock: null as () => void,
-  onUnload: null as () => void,
-  config: defaultConfig,
-}
-
-class ExternalWindow extends React.Component<ExternalWindowProps & { platform: PlatformAdapter }> {
-  externalWindow: Window | null = null
-
-  async componentDidMount() {
-    const { config, platform } = this.props
+const ExternalWindow: FC<ExternalWindowProps> = ({
+  title = '',
+  onBlock = null as () => void,
+  onUnload = null as () => void,
+  config = defaultConfig,
+}) => {
+  const platform = usePlatform()
 
-    this.externalWindow = await platform.window.open(config, this.release)
+  let externalWindow: Window
 
-    if (this.externalWindow) {
-      window.addEventListener('beforeunload', this.release)
+  const release = () => {
```

Comment:
declaring `release` like this on a functional component will overwrite the reference to the function on every render, so by the time we do
```
window.removeEventListener('beforeunload', release)
```
below, `release` could no longer be the same function we added the event listener to. See [this](https://stackoverflow.com/questions/53845595/wrong-react-hooks-behaviour-with-event-listener) similar issue on stackoverflow.

When using hooks we may want to use something like [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback).

Reply:
Good catch.